anomalous-value/src/agent.js

```diff
@@ -40,45 +28,97 @@ function createAlert(log) {
   });
 }
 
-async function handleTransaction(txEvent) {
-  const findings = [];
-
-  const parsedLogs = txEvent.filterLog(eventFragments, address);
-
-  // loop over each eventLog, checking for anomalous value
-  parsedLogs.forEach((log) => {
-    // rolling math requires bignumber.js style BigNumbers
-    const amount = new BigNumber(log.args.amount.toHexString());
-
-    // if we haven't seen this reserve yet, initialize it
-    if (!rollingEventData[log.args.reserve]) {
-      rollingEventData[log.args.reserve] = new RollingMath(anomalousValue.windowSize);
-    }
-
-    // only process data for alerts if we have seen a significant number of blocks
-    if (rollingEventData[log.args.reserve].getNumElements() >= anomalousValue.windowSize) {
-      // if we have seen this before, check for anomalous value
-      const average = rollingEventData[log.args.reserve].getAverage();
-      const standardDeviation = rollingEventData[log.args.reserve].getStandardDeviation();
+function provideInitialize(data) {
+  return async function initialize() {
+    /* eslint-disable no-param-reassign */
+    const { LendingPool: lendingPoolConfig } = config.contract;
+    const abi = getAbi(lendingPoolConfig.abiFile);
+
+    // create ethers interface object
+    const iface = new ethers.utils.Interface(abi);
+
+    // events we are interested in
+    const eventFragments = [
+      iface.getEvent('Borrow'),
+      iface.getEvent('Deposit'),
+      iface.getEvent('Repay'),
+      iface.getEvent('Withdraw'),
+    ];
```

Comment:
On one hand I like that it is explicitly stated what events are being monitored here, but this diverges from our more recent approach of putting these under the `events` key in the `bot-config.json` file.

I'm fine with it being left this way, given how specific the Bot is, but it seemed worth mentioning.

Reply:
Agreed, this is in-line with my previous comment about investigating how to generalize this bot.